diff --git a/Include/opcode.h b/Include/opcode.h
index be360e1..c056801 100644
--- a/Include/opcode.h
+++ b/Include/opcode.h
@@ -126,6 +126,7 @@ extern "C" {
 #define BUILD_CONST_KEY_MAP     156
 #define BUILD_STRING            157
 #define BUILD_TUPLE_UNPACK_WITH_CALL 158
+#define CALL_UNARY_FUNCTIONS    200
 
 /* EXCEPT_HANDLER is a special, implicit block type which is created when
    entering an except handler. It is not an opcode but we define it here
diff --git a/Python/ceval.c b/Python/ceval.c
index 9ad582b..372c5b7 100644
--- a/Python/ceval.c
+++ b/Python/ceval.c
@@ -3516,6 +3516,21 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
             goto dispatch_opcode;
         }
 
+        TARGET(CALL_UNARY_FUNCTIONS) {
+            for (Py_ssize_t i = 0; i < oparg; i++) {
+                PyObject **sp, *res;
+                sp = stack_pointer;
+                res = call_function(&sp, 1, NULL);
+                stack_pointer = sp;
+                PUSH(res);
+                if (res == NULL) {
+                    goto error;
+                }
+            }
+
+            DISPATCH();
+        }
+
 
 #if USE_COMPUTED_GOTOS
         _unknown_opcode:
